### PR TITLE
restructure: move RAGapp code to src/ragapp

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: deployments/multiple-ragapps/docker-compose.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ services:
       - ragapp-network
   
   manager:
-    image: ragapp/manager
+    image: ragapp/manager:latest
```
Comment (marcusschiesser) [inline deployments/multiple-ragapps/docker-compose.yml:68]: build update for ragapp missing?

File: README.md [hunk 2]
```diff
@@ -64,15 +64,18 @@ You can easily deploy RAGapp to your own infrastructure with one of these Docker
...
 export ENVIRONMENT=dev
 poetry install --no-root
 make build-frontends
 make dev
 ```
 
-> _Note_: To check out the admin UI during development, please go to http://localhost:3000/admin.
+> _Note_:  
+Make sure you have the [Poetry](https://python-poetry.org/) installed.  
+To check out the admin UI during development, please go to http://localhost:3000/admin.
```
Comment (marcusschiesser) [inline README.md:78]:
```suggestion
Make sure you have [Poetry](https://python-poetry.org/) installed.  
To check out the admin UI during development, please go to http://localhost:3000/admin.
```

File: .github/workflows/manager_docker_images.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            ragapp/manager:${{ steps.tag.outputs.TAG }}
+            ragapp/manager:latest
+
+      - name: Create GitHub release
+        uses: ncipollo/release-action@v1
+        with:
+          artifacts: "README.md"
+          name: Release ${{ steps.tag.outputs.TAG }}
+          bodyFile: "CHANGELOG.md"
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (marcusschiesser) [inline .github/workflows/manager_docker_images.yml:55]: don't we end up having two github releases now?

File: .github/workflows/manager_docker_images.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    paths:
+      - 'src/manager/**'
+
+jobs:
+  release_tagged_image:
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: "src/manager"
```
Comment (marcusschiesser) [inline .github/workflows/manager_docker_images.yml:18]: can we reuse the same process for both dirs? src/manager and src/ragapp?

Comment (marcusschiesser) [reply .github/workflows/manager_docker_images.yml:18]: (can do this in a later PR)

File: .github/workflows/lint_on_push_or_pull.yml [hunk 5]
```diff
@@ -49,15 +73,14 @@ jobs:
...
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          version: ${{ env.POETRY_VERSION }}
-
-      - name: Install pre-commit
+      - name: Install black
         shell: bash
-        run: poetry install --with dev
```
Comment (marcusschiesser) [inline .github/workflows/lint_on_push_or_pull.yml:59]: TODO (for next PR: make new workflow, installing poetry for each python package and run unit tests)

File: deployments/multiple-ragapps/docker-compose.yml (hunk 1 above)
Comment (leehuwuj) [reply deployments/multiple-ragapps/docker-compose.yml:68]: hmm, they all have the build config

File: .github/workflows/manager_docker_images.yml (hunk 3 above)
Comment (leehuwuj) [reply .github/workflows/manager_docker_images.yml]: I just updated the pipeline for tag build with pattern: `ragapp-v` or `manager-v`. Please help me to have a look.

File: deployments/multiple-ragapps/docker-compose.yml (hunk 1 above)
Comment (marcusschiesser) [reply deployments/multiple-ragapps/docker-compose.yml:68]: the build path for ragapp container is not changed

File: .github/workflows/manager_docker_images.yml [hunk 6]
```diff
@@ -5,17 +5,15 @@ on:
     branches:
       - main
     tags:
-      - v*
+      - manager-v*
```
Comment (marcusschiesser) [inline .github/workflows/manager_docker_images.yml:8]: we should use the same tag for both images, just push a `v*` tag and both images must be build

File: .github/workflows/manager_docker_images.yml (hunk 3 above)
Comment (marcusschiesser) [reply .github/workflows/manager_docker_images.yml]: that's not good, we need one release `v*`

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**